src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/ModuleInitializerAttributeShouldNotBeUsedInLibrariesTests.cs

```diff
@@ -0,0 +1,254 @@
+            await new VerifyCS.Test
+            {
+                TestState =
+                {
+                    ReferenceAssemblies = ReferenceAssemblies.Net.Net50,
+                    Sources = {
+                        @$"
+                        {classModifier} class AccessibleClass
+                        {{
+                            [[|System.Runtime.CompilerServices.ModuleInitializer|]]
+                            {methodModifier} static {(useAsync ? "async" : "")} void AccessibleInitializer() {{ }}
+                        }}
+                        "
+                    }
+                },
+                LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp9
+            }.RunAsync();
+        }
+
+        [Theory]
+        [InlineData("public", "public", false)]
+        [InlineData("public", "internal", false)]
+        [InlineData("internal", "public", false)]
+        [InlineData("internal", "internal", false)]
+        [InlineData("public", "public", true)]
+        [InlineData("public", "internal", true)]
+        [InlineData("internal", "public", true)]
+        [InlineData("internal", "internal", true)]
+        public async Task CA2255ModuleInitializerOnMethod_WithParens(string classModifier, string methodModifier, bool useAsync)
+        {
+            await new VerifyCS.Test
+            {
+                TestState =
+                {
+                    ReferenceAssemblies = ReferenceAssemblies.Net.Net50,
+                    Sources = {
+                        @$"
+                        {classModifier} class AccessibleClass
+                        {{
+                            [[|System.Runtime.CompilerServices.ModuleInitializer()|]]
+                            {methodModifier} static {(useAsync ? "async" : "")} void AccessibleInitializer() {{ }}
+                        }}
+                        "
+                    }
+                },
+                LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp9
+            }.RunAsync();
+        }
+
+        [Theory]
+        [InlineData("public", "public", false)]
+        [InlineData("public", "internal", false)]
+        [InlineData("internal", "public", false)]
+        [InlineData("internal", "internal", false)]
+        [InlineData("public", "public", true)]
+        [InlineData("public", "internal", true)]
+        [InlineData("internal", "public", true)]
+        [InlineData("internal", "internal", true)]
+        public async Task CA2255ModuleInitializerOnMethod_Suppressed(string classModifier, string methodModifier, bool useAsync)
+        {
+            await new VerifyCS.Test
+            {
+                TestState =
+                {
+                    ReferenceAssemblies = ReferenceAssemblies.Net.Net50,
+                    Sources = {
+                        @$"
+                        {classModifier} class AccessibleClass
+                        {{
+                        #pragma warning disable CA2255
```

Comment:
NIT: we don't really need to test suppression, but i am ok with keeping it

Reply:
I added it for 2 reasons:

1. To validate the diagnostic is `CA2255` as expected (that doesn't get validated anywhere else)
2. Since the analyzer first identifies the method and then adds the diagnostic to the attribute, I wanted to illustrate the suppression only needs to be around the attribute (not the method)